sql/core/src/test/scala/org/apache/spark/sql/DataFrameComplexTypeSuite.scala

```diff
@@ -64,6 +70,33 @@ class DataFrameComplexTypeSuite extends QueryTest with SharedSparkSession {
     val ds100_5 = Seq(S100_5()).toDS()
     ds100_5.rdd.count
   }
+
+  test("SPARK-29503 nest unsafe struct inside safe array") {
+    withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false") {
```

Comment:
This issue is encountered only when whole stage codegen is disabled?

Reply:
At least yes for provided reproducer. For other case I'm not sure.

Reply:
I see. For whole stage codegen, CreateNamedStruct is not converted to CreateNamedStructUnsafe, so the nested struct is not unsafe one.

Reply:
Ah OK got it. Thanks for explanation.